### ISSUE TITLE
Support for neovim/vim

### ISSUE THREAD

**Issue by Hamza12700** [DanielXMoore/Civet#1320]
I recently discovery this project and I really like the syntax of the language. So I was thinking to give it a shot but unfortunately the language server only supports VSCode. It'd be really nice if the language server also supported neovim/vim.

**Comment by edemaine** [comment]
Agreed! Someone needs to spend the time to separate the LSP process from the rest of the VSCode Plugin. I believe they are mostly separate already, and it's mostly a distribution/bundling issue. I don't *think* it should be hard (assuming LSP protocols are the same), but only trying will tell. 😺   Are you interested in giving it a try?

**Comment by Hamza12700** [comment]
Yeah, I can give it a shot

**Comment by miloslavnosek** [comment]
Any news on this? It would really help with enjoyment of this language to not be editor-locked.

**Comment by STRd6** [comment]
I looked into this briefly but ran into ran into some trouble because I have no experience configuring neovim.

It seems that something similar to this should work though:

```
vim.lsp.config('civet_lsp', {
  cmd = { 'node', '~/Downloads/DanielX.civet-0.3.28/extension/dist/server.js', '--stdio' },
  filetypes = { 'civet' },
  root_dir = util.root_pattern('tsconfig.json', 'package.json', '.git'),
  single_file_support = true,
})
```

This is assuming that the published VSCode extension is downloaded and extracted. https://marketplace.visualstudio.com/_apis/public/gallery/publishers/DanielX/vsextensions/civet/0.3.28/vspackage

I didn't quite get it working but maybe this will encourage someone who has more experience with neovim to give it a shot and eventually we can get over the finish line.

**Comment by STRd6** [comment]
Might be able to mark this as done: https://civet.dev/integrations#neovim

There's also a basic treesitter grammar as well.
